### ISSUE TITLE
[Question] Why is PPO loss needed?

### ISSUE THREAD

**Issue by JoaoLages** [lvwerra/trl#84]
First of all, thank you for the great package 💪, I just wished that we could make it work for enc-dec models like T5!
My question is: why is the [PPO loss ](https://github.com/lvwerra/trl/blob/7ed040fb23b7f6c6db0077aea3579a0bbabf6947/trl/trainer/ppo_trainer.py#L493)so complicated? Couldn't we just train with `loss = reward + KL`?

**Comment by lvwerra** [comment]
the reward itself is not differentiable and as such you can't backpropagate it. PPO is one way of estimating a differentiable loss from a non-differentiable reward.

**Comment by JoaoLages** [comment]
Ah yes, nicely noted. The main differentiable output is the ratio between new log probabilities and initial log probabilities, right? `log_probs / old_log_probs`
